web/src/src/__tests__/recipes-create.tsx

```diff
@@ -1,49 +1,31 @@
 import '@testing-library/jest-dom'
-import axios from 'axios'
 import React from 'react'
 import { fireEvent, screen, waitForElement, render, cleanup } from '@testing-library/react'
 import App from '../App'
-
+import {RecipeService} from '../recipe-service/recipe-service'
+import { act } from 'react-dom/test-utils'
 const recipe = {
   name: "pizza",
   id: 1,
   description: "desc",
   ingredients: [{ name: "cheese" }]
 }
 
-const originalError = console.error
-beforeAll(() => {
-  jest.spyOn(window, 'alert').mockImplementation(() => { });
-  console.error = (...args) => {
-    if (/Warning.*not wrapped in act/.test(args[0])) {
-      return
-    }
-    originalError.call(console, ...args)
-  }
-})
-
-afterAll(() => {
-  console.error = originalError
-})
-
-afterEach(() => {
-  cleanup()
-})
 
 test('must add ingredients', async () => {
-  jest.spyOn(axios, 'get').mockImplementation(() => Promise.resolve({ data: [recipe] }))
-  jest.spyOn(axios, 'post').mockImplementation(() => Promise.resolve({ data: {} }))
-  render(<App />)
+  jest.spyOn(RecipeService, 'list').mockImplementation(() => Promise.resolve([recipe]))
+  jest.spyOn(RecipeService, 'save').mockImplementation(() => Promise.resolve({}))
+  
+  await act(async () => {render(<App />) })
```

Comment:
why this await and act?

Reply:
I was receiving a warning because of `useEffect`
The solution I came across was based on [this](https://stackoverflow.com/questions/55047535/testing-react-components-that-fetches-data-using-hooks)